ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/command/AbstractRatisCommand.java

```diff
@@ -0,0 +1,179 @@
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * The base class for all the ratis shell {@link Command} classes.
+ */
+public abstract class AbstractRatisCommand implements Command {
+  public static final String PEER_OPTION_NAME = "peers";
+  public static final String GROUPID_OPTION_NAME = "groupid";
+  public static final RaftGroupId DEFAULT_RAFT_GROUP_ID
+      = RaftGroupId.valueOf(UUID.fromString("1-1-1-1-1"));
+  private final PrintStream printStream;
+  private RaftGroup raftGroup;
+
+  protected AbstractRatisCommand(Context context) {
+    printStream = context.getPrintStream();
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    List<InetSocketAddress> addresses = new ArrayList<>();
+    String peersStr = "";
+    if (cl.hasOption(PEER_OPTION_NAME)) {
+      peersStr = cl.getOptionValue(PEER_OPTION_NAME);
+    }
+    String[] peersArray = peersStr.split(",");
+    for (int i = 0; i < peersArray.length; i++) {
+      String[] hostPortPair = peersArray[i].split(":");
+      InetSocketAddress addr =
+          new InetSocketAddress(hostPortPair[0], Integer.parseInt(hostPortPair[1]));
+      addresses.add(addr);
+    }
+
+    RaftGroupId raftGroupIdFromConfig = DEFAULT_RAFT_GROUP_ID;
+    if (cl.hasOption(GROUPID_OPTION_NAME)) {
+      raftGroupIdFromConfig = RaftGroupId.valueOf(
+          UUID.fromString(cl.getOptionValue(GROUPID_OPTION_NAME)));
+    }
+
+    List<RaftPeer> peers = addresses.stream()
+        .map(addr -> RaftPeer.newBuilder()
+            .setId(RaftUtils.getPeerId(addr))
+            .setAddress(addr)
+            .build()
+        ).collect(Collectors.toList());
+    raftGroup = RaftGroup.valueOf(raftGroupIdFromConfig, peers);
+    try (final RaftClient client = RaftUtils.createClient(raftGroup)) {
+      RaftGroupId remoteGroupId;
+      List<RaftGroupId> groupIds;
+      groupIds = RetryUtil.run(peers,
+          p -> {
+            try {
+              return client.getGroupManagementApi((p.getId())).list().getGroupIds();
+            } catch (IOException e) {
+              e.printStackTrace();
+              return null;
+            }
+          }
+      );
+
+      if (groupIds.size() == 1) {
+        remoteGroupId = groupIds.get(0);
+      } else {
+        final UUID raftGroupUuid = raftGroupIdFromConfig.getUuid();
+        Optional<RaftGroupId> raftGroupId =
+            groupIds.stream().filter(r -> raftGroupUuid.equals(r.getUuid()))
+                .findFirst();
+        if (!raftGroupId.isPresent()) {
+          printStream.println(
+              "there are more than one group, you should specific one."
+                  + groupIds);
+          return -1;
+        } else {
+          remoteGroupId = raftGroupId.get();
+        }
+      }
+      raftGroup = RetryUtil.run(peers,
+          p -> {
+            try {
+              return client.getGroupManagementApi((p.getId()))
+                  .info(remoteGroupId).getGroup();
```

Comment:
We should store the group info reply so that it can be reused in InfoCommand.

Reply:
Sure